### DISCUSSION TITLE
Can´t apply custom x y values to plotly chart

### DISCUSSION THREAD

**Discussion by falkio8** [dbuezas/lovelace-plotly-graph-card#462] [Q&A] [answered]
I try to display custom x and y values by a plotly graph. The values are stored as attributes in a sensor:

![grafik](https://github.com/user-attachments/assets/acd6d8f7-7b9d-40f4-937a-29a5226f39f2)

template tester delivers the correct string:
![grafik](https://github.com/user-attachments/assets/d9ac68f1-e6b4-4d4e-8ea3-f0c2d03dcb69)

They are stored in attributes because of the 256bit limitation of sensor values.

When I manually apply the values to the plotly graph it works:

    entity: ‘’
    name: test
    line:
    shape: spline
    color: white
    x: [-20, -19, -18, -17, -16, -15, -14, -13, -12, -11, -10, -9, -8, -7, -6, -5, -4, -3, -2, -1, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]
    ‘y’: [55, 55, 55, 55, 55, 55, 55, 55, 55, 55, 55, 55, 55, 55, 55, 55, 54.24, 53.21, 52.16, 51.09, 50, 48.89, 47.76, 46.61, 45.44, 44.25, 43.04, 41.81, 40.56, 39.29, 38, 36.69, 35.36, 34.01, 32.64, 31.25, 29.84, 28.41, 26.96, 25.49, 24]

![grafik](https://github.com/user-attachments/assets/8bb07962-fe61-4f69-9ecc-8c97472eb294)

but trying to access the attributes and hand them over to the entity does result in an empty chart. What I tried so far:

        x: >
          {{ states['sensor.trovis_regler_vorlaufkurve'].attributes.y_werte
          }}

or

x: "[[[ return states['sensor.trovis_regler_vorlaufkurve'].attributes.x_werte; ]]]"
'y': "[[[ return states['sensor.trovis_regler_vorlaufkurve'].attributes.y_werte; ]]]"

or

    x: |-
      [[[
        return states['sensor.trovis_regler_vorlaufkurve'].attributes.x_werte;
      ]]]
    'y': |-
      [[[
        return states['sensor.trovis_regler_vorlaufkurve'].attributes.y_werte;
      ]]]

![grafik](https://github.com/user-attachments/assets/a3ec4b16-4356-4b29-921d-b24d5be9cbe4)

I don´t get why plotly is not able to “read” the applied values. For the last option I know that this works for another guy! Appreciate any help!

**Comment by dbuezas** [comment, marked as answer]
Double curly braces templating is not available in the configuration of cards, and triple square brackets are another syntax that requires an extra templating extension. 

In this card there's no templating, there are JavaScript functions instead.

Try with this instead:

```
x: |
   $ex hass.states['sensor.trovis_regler_vorlaufkurve'].attributes.x_werte
y: |
   $ex hass.states['sensor.trovis_regler_vorlaufkurve'].attributes.y_werte
    
```

**Reply by falkio8** [reply]
OMG tinkering for 3 days and now it works, thank you!

![image](https://github.com/user-attachments/assets/7ed04098-0029-42dc-9d73-b46dd47b7f62)

**Reply by dbuezas** [reply]
Great!
Try out the yaml editor linked in the readme.
It won't fix JavaScript, but would have told you something was wrong with what you tried
